packages/suite/src/utils/wallet/coinmarket/buyUtils.ts

```diff
@@ -79,26 +78,28 @@ export function createQuoteLink(request: BuyTradeQuoteRequest, account: Account)
     const params = `offers/${account.symbol}/${account.accountType}/${account.index}/${hash}`;
 
     if (isDesktop()) {
-        return `${ELECTRON_RECEIVER_SERVER}/buy-redirect?p=${encodeURIComponent(
+        const serverUrl = await getElectronReceiverUrl();
+        return `${serverUrl}/buy-redirect?p=${encodeURIComponent(
             `/coinmarket-redirect/${params}`,
         )}`;
     }
 
     return `${locationOrigin}${assetPrefix}/coinmarket-redirect#${params}`;
-}
+};
 
-export function createTxLink(trade: BuyTrade, account: Account): string {
+export const createTxLink = async (trade: BuyTrade, account: Account) => {
     const locationOrigin = getLocationOrigin();
     const assetPrefix = process.env.assetPrefix || '';
     const params = `detail/${account.symbol}/${account.accountType}/${account.index}/${trade.paymentId}`;
     if (isDesktop()) {
-        return `${ELECTRON_RECEIVER_SERVER}/buy-redirect?p=${encodeURIComponent(
+        const serverUrl = await getElectronReceiverUrl();
```

Comment:
the goal in this issue is to get rid of `if (desktop)` conditions from this file and use the same request url (line 101)
and do the "electron opening custom local server" logic on the electron side.

currently you've changed ELECTRON_RECEIVER_SERVER for getElectronReceiverUrl but this is not what we want here

Reply:
But we are not handling the request, we are just passing it to the partner. We can't remove `if (desktop)` from here.

Reply:
but that was the whole idea, to remove it. see issue description. discussed together with @martinboehm and you in person